### PR TITLE
[Fix #10585] Add http-client error log for non-CompletableFuture responses

### LINKED ISSUE #10585: Implement http-client error log for non-CompletableFuture responses
### Discussed in https://github.com/micronaut-projects/micronaut-core/discussions/10551

<div type='discussions-op-text'>

<sup>Originally posted by **stefanos-kalantzis** February 27, 2024</sup>
Case in point: https://github.com/micronaut-projects/micronaut-core/blob/3fc45e7e70982cd1b9ef104c03053246e70ce2c5/http-client-core/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java#L384</div>

### PR DESCRIPTION
**Pull request by glorrian**
Add logging for sync and publisher client types as mentioned in [#10585](https://github.com/micronaut-projects/micronaut-core/issues/10585).

### CODE REVIEW

**Comment by dstepanov** [comment]
Needs a test

**Comment by glorrian** [comment]
@dstepanov i've improved the code and added tests

File: http-client-core/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java [hunk 1]
```diff
@@ -332,6 +337,14 @@ private Object handlePublisher(MethodInvocationContext<Object, Object> context,
         } else {
             publisher = httpClientResponsePublisher(httpClient, requestPublisher, returnType, errorType, valueType);
         }
+
+        publisher = Flux.from(publisher)
```
Comment (yawkat) [inline http-client-core/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java]: make this entire statement conditional on isDebugEnabled

Comment (glorrian) [reply http-client-core/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java]: done

**Review by yawkat — APPROVED** [review]

**Review by dstepanov — APPROVED** [review]

**Comment by yawkat** [comment]
Thanks!

**Pull request merged**